src/csm_testing/lib/iscsi_common.py

```diff
@@ -0,0 +1,289 @@
+    Returns:
+        stdout/stderr(str),returncode(int)
+    """
+    try:
+        result = subprocess.run(
+            command,
+            shell=True,
+            check=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            universal_newlines=True,
+        )
+        return result.stdout.strip(), result.returncode
+    except subprocess.CalledProcessError as err:
+        if err.stderr.strip():
+            return err.stderr.strip(), err.returncode
+        return err.stdout.strip(), err.returncode
+
+
+def compare_versions(version1, version2):
+    """
+    Function to compare given two versions and print the result info.
+    Args:
+        version1(int/float), version2(int/float) : versions to compare
+    Returns:
+        Bool
+    """
+    ver_1 = version.parse(version.Version(version1).base_version)
+    ver_2 = version.parse(version2)
+
+    if ver_1 < ver_2:
+        return False
+    return True
+
+
+def iscsi_prechecks_failed():
+    """
+    Function to handle exit and print the ERROR info.
+    Args:
+        None
+    Returns:
+        None
+    """
+    print("ERROR: iSCSI pre checks Failed")
+    print(
+        "-----------------------END OF iSCSI PRE-CHECKS--------------------------------"
+    )
+    sys.exit(1)
+
+def iscsi_sbps_failed(xname=None):
+    """
+    Function to handle exit and print the ERROR info.
+    Args:
+        xname(str)
+    Returns:
+        None
+    """
+    if xname is not None:
+        print(f"ERROR: iSCSI SBPS Readiness Test Failed on Worker Node {xname}")
+        print(
+            f"---------------------END OF iSCSI SBPS Readiness Test on {xname}------------------"
+        )
+    else:
+        print("ERROR: iSCSI SBPS Readiness Test Failed")
+        print(
+            "-----------------------END OF iSCSI SBPS Readiness Test--------------------"
+        )
+    sys.exit(1)
+
+def check_ssh_xname(xname):
+    """
+    Function to check ssh connectivity and print the info.
+    Args:
+        xname(string)
+    Returns:
+        Bool
+    """
+    ssh_command = f"ssh -o BatchMode=yes -q {xname} exit"
+    ssh_output, returncode = run_command(ssh_command)
+
+    if returncode == 0:
+        return True
+    print(f"ERROR: Unable to SSH to {xname}")
+    return False
+
+
+def get_compute_node() -> list:
+    """
+    Function to get list of stable compute nodes.
+    Args:
+        None
+    Returns:
+        Compute Nodes List (list)
+    """
+    #Get List of configured compute nodes
+    sat_status_output, returncode = run_command(
+        command="""sat status --filter 'role=compute and \
+               State=Ready and \"Configuration Status\"=configured ' \
+               --fields xname  --no-borders --no-headings"""
+    )
+    if returncode != 0:
+        print(f"ERROR: Command sat status returned error. Return code: {returncode}")
+        iscsi_prechecks_failed()
+    elif sat_status_output == "":
```

Comment:
Because the previous conditional branch always exits, no need to use `elif` here.
```suggestion
    if sat_status_output == "":
```